frontend/src/liboperator/operator/operator.go

```diff
@@ -533,60 +534,49 @@ func adbProxy(w http.ResponseWriter, r *http.Request, pool *DevicePool) {
 	defer tcpConn.Close()
 
 	// Redirect WebSocket to ADB tcp socket
-	redirectWsToTcp(wsConn, tcpConn)
-}
-
-func redirectWsToTcp(wsConn *websocket.Conn, tcpConn net.Conn) {
-	tcpChan := make(chan []byte)
-	wsChan := make(chan []byte)
-
-	go func(tcpConn net.Conn, tcpChan chan []byte) {
-		buf := make([]byte, 4096)
-		for {
-			nRead, err := tcpConn.Read(buf)
-
-			if err != nil {
-				close(tcpChan)
-				return
-			}
-
-			tcpData := make([]byte, nRead)
-			copy(tcpData, buf)
-
-			tcpChan <- tcpData
-		}
-	}(tcpConn, tcpChan)
+	wsWrapper := &wsIoWrapper{
+		wsConn: wsConn,
+		pos: 0,
+		buf: nil,
+	}
 
-	go func(wsConn *websocket.Conn, wsChan chan []byte) {
-		for {
-			_, buf, err := wsConn.ReadMessage()
+	go io.Copy(wsWrapper, tcpConn)
+	io.Copy(tcpConn, wsWrapper)
+}
 
-			if err != nil {
-				close(wsChan)
-				return
-			}
+// Wrapper for implementing io.ReadWriteCloser of websocket.Conn
+type wsIoWrapper struct {
+	wsConn *websocket.Conn
+	pos int
+	buf []byte
+}
 
-			wsData := make([]byte, len(buf))
-			copy(wsData, buf)
+var _ io.ReadWriteCloser = (*wsIoWrapper)(nil)
```

Comment:
why is this needed?

Reply:
This code is for confirming that `wsIoWrapper` implements `io.ReadWriteCloser` interface

Reply:
I see, from https://go.dev/doc/effective_go#interfaces_and_types this is needed when interface checks happens at runtime which is not the case here. 

```
The appearance of the blank identifier in this construct indicates that the declaration exists only for the type checking, not to create a variable. Don't do this for every type that satisfies an interface, though. By convention, such declarations are only used when there are no static conversions already present in the code, which is a rare event.
```